src/routes/download/drs.service.ts

```diff
@@ -0,0 +1,70 @@
+import urlJoin from 'url-join';
+import fetch from 'node-fetch';
+
+import { AppConfig } from '@/config/global';
+import logger from '@/logger';
+
+// types from https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/#_accessmethod
+// will only be using https
+enum AccessType {
+  HTTPS = 'https',
+  S3 = 's3',
+  GS = 'gs',
+  FTP = 'ftp',
+  FILE = 'file',
+  GSIFTP = 'gsiftp',
+  GLOBUS = 'globus',
+  HTSGET = 'htsget',
+}
+
+interface AccessMethod {
+  type: AccessType;
+  access_url: { url: string };
+  region: string;
+}
+
+export interface DRSMetada {
+  name: string;
+  accessUrls: AccessMethod[];
+}
+
+export const parseDRSPath = (filepath: string, config: AppConfig) => {
```

Comment:
This doesnt need the whole config... just drsProtocol and drsObjectPath. Really these should just be imported directly from Config and not passed into the method.

Reply:
i did this cuz i didn't want to make the function async...but i can change it

Reply:
see my comment at the end of this PR, I anticipated this 🔮 🧙

Reply:
esp now that i've read your comment about separating out the config function 🙂